src/services/NumberService.ts

```diff
@@ -141,6 +141,38 @@ export class NumberService {
             });
     }
 
+    /**
+     * Creates a new plate serial number in the database.
+     * @param attempt - the current number of attempts for generating a Plate Serial Number.
+     * @param awsError - AWS error to be passed when the request fails otherwise null.
+     */
+    public createPlateSerialNumber(attempts: number, awsError: AWSError | null): Promise<PlateSerialNumber> {
+        this.manageAttempts(attempts, awsError);
+        return this.getLastPlateSerialNumber()
+            .then((lastPlateSerialNumber) => {
+                const nextPlateSerialNumberObject = this.createNextPlateSerialNumberObject(lastPlateSerialNumber);
+                const transactExpression = {
+                    ConditionExpression: "plateSerialNumber = :oldPlateSerialNumber",
+                    ExpressionAttributeValues: {
+                        ":oldPlateSerialNumber": lastPlateSerialNumber.plateSerialNumber
+                    }
+                };
+                return this.dbClient.transactWrite(nextPlateSerialNumberObject, transactExpression)
+                    .then(() => {
+                        console.log(`Plate Serial Number Generated successfully`);
+                        return nextPlateSerialNumberObject;
+                    })
+                    .catch((error: AWSError) => {
+                        console.error(error); // limit to 5 attempts
+                        if (error.statusCode === 400) {
+                            console.error(`Attempt number ${attempts} failed. Retrying up to ${Configuration.getInstance().getMaxAttempts()} attempts.`);
```

Comment:
Modify the string message to represent the type of number getting generated e.g. 
`Attempt number 2 for trailerId failed`

Reply:
done